meta not indicated as reserved crate name when shadowing module

Taken from this internals thread: https://internals.rust-lang.org/t/is-the-module-name-meta-forbidden/9587

A module named `meta` is shadowed leading to misleading error messages. `meta` was reserved as a built-in crate name in 2018 edition, but the error message does not indicate this.

I tried this code:

```rust
mod meta {
    pub struct Thing;
}

use meta::Thing;

fn main() {}
```
Above code compiled with `rustc --edition 2018 {file}`

This is expected to indicated that `meta` is a reserved crate name.

Instead, the following error is reported:
```
error[E0463]: can't find crate for `meta`
 --> testy.rs:5:5
  |
5 | use meta::Thing;
  |     ^^^^ can't find crate

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
```

### Meta
`rustc --version --verbose`:
```
rustc 1.44.1 (c7087fe00 2020-06-17)
binary: rustc
commit-hash: c7087fe00d2ba919df1d813c040a5d47e43b0fe7
commit-date: 2020-06-17
host: x86_64-unknown-linux-gnu
release: 1.44.1
LLVM version: 9.0
```


<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->
This issue has been assigned to @chocol4te via [this comment](https://github.com/rust-lang/rust/issues/73948#issuecomment-654795936).
<!-- TRIAGEBOT_ASSIGN_DATA_START$${"user":"chocol4te"}$$TRIAGEBOT_ASSIGN_DATA_END -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/compiler/rustc_resolve/src/lib.rs
+++ b/compiler/rustc_resolve/src/lib.rs
@@ -1240,9 +1240,6 @@ impl<'a> Resolver<'a> {
             extern_prelude.insert(Ident::with_dummy_span(sym::core), Default::default());
             if !session.contains_name(&krate.attrs, sym::no_std) {
                 extern_prelude.insert(Ident::with_dummy_span(sym::std), Default::default());
-                if session.rust_2018() {
-                    extern_prelude.insert(Ident::with_dummy_span(sym::meta), Default::default());
-                }
             }
         }
 

--- a/src/test/ui/rfc-2126-extern-absolute-paths/meta.rs
+++ b/src/test/ui/rfc-2126-extern-absolute-paths/meta.rs
@@ -1,7 +1,0 @@
-// edition:2018
-
-// Tests that `meta` is allowed, even if the crate doesn't exist
-// yet (i.e., it causes a different error than `not-allowed.rs`).
-use meta; //~ ERROR can't find crate for `meta`
-
-fn main() {}

--- a/src/test/ui/rfc-2126-extern-absolute-paths/meta.stderr
+++ b/src/test/ui/rfc-2126-extern-absolute-paths/meta.stderr
@@ -1,9 +1,0 @@
-error[E0463]: can't find crate for `meta`
-  --> $DIR/meta.rs:5:5
-   |
-LL | use meta;
-   |     ^^^^ can't find crate
-
-error: aborting due to previous error
-
-For more information about this error, try `rustc --explain E0463`.

